### PR TITLE
various bugfixes

### PR DESCRIPTION
**Pull request by croqueGrec09**
includes commits for:
- ERMS-2527: subscription copy bugfixes and improvements
- no ticket: global source sync transactions included

### CODE REVIEW

File: app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy [hunk 1]
```diff
@@ -4713,12 +4713,13 @@ class SubscriptionController extends AbstractDebugController {
...
                         AuditConfig.addConfig(newSub, auditField)
                     }
+
                 }
                 //Copy References
                 //OrgRole
                 baseSub.orgRelations?.each { or ->
 
-                    if ((or.org?.id == contextService.getOrg().id) || (or.roleType in [RDStore.OR_SUBSCRIBER, RDStore.OR_SUBSCRIBER_CONS])) {
+                    if ((or.org.id == contextService.getOrg().id) || (or.roleType in [OR_SUBSCRIBER, OR_SUBSCRIBER_CONS])) {
```
Comment (klodav) [inline app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy:4722]: for clarity; please use not **import static de.laser.helper.RDStore.***

File: app/grails-app/services/de/laser/SubscriptionService.groovy [hunk 2]
```diff
@@ -151,12 +151,7 @@ class SubscriptionService {
     }
 
     List getValidSubChilds(Subscription subscription) {
-        List<Subscription> validSubChildren = Subscription.findAllByInstanceOf(subscription)
-        validSubChildren = validSubChildren?.sort { Subscription a, Subscription b ->
-            Org sa = a.getSubscriber()
-            Org sb = b.getSubscriber()
-            (sa.sortname ?: sa.name ?: "")?.compareTo((sb.sortname ?: sb.name ?: ""))
-        }
+        List<Subscription> validSubChildren = Subscription.executeQuery('select oo.sub from OrgRole oo where oo.sub.instanceOf = :sub and oo.roleType in (:subRoleTypes) order by oo.org.sortname asc, oo.org.name asc',[sub:subscription,subRoleTypes:[OR_SUBSCRIBER_CONS,OR_SUBSCRIBER,OR_SUBSCRIBER_CONS_HIDDEN]])
```
Comment (klodav) [inline app/grails-app/services/de/laser/SubscriptionService.groovy:154]: for clarity; please use not **import static de.laser.helper.RDStore.***

File: app/grails-app/services/de/laser/SubscriptionService.groovy [hunk 3]
```diff
@@ -151,12 +151,7 @@ class SubscriptionService {
...
         validSubChildren
```
Comment (anjaalbinnrw) [inline app/grails-app/services/de/laser/SubscriptionService.groovy:155]: :-)

Comment (croqueGrec09) [reply app/grails-app/services/de/laser/SubscriptionService.groovy:155]: Sorgt vor allem nicht mehr für Abstürze! :)

File: app/grails-app/services/de/laser/SubscriptionService.groovy (hunk 2 above)
Comment (croqueGrec09) [reply app/grails-app/services/de/laser/SubscriptionService.groovy]: Changed in f79154e

File: app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy (hunk 1 above)
Comment (croqueGrec09) [reply app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy]: changed in f79154e

**Pull request merged**